addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -105,30 +105,28 @@ var SnippetEditor = Widget.extend({
             $customize.find('.oe_snippet_remove').addClass('d-none');
         }
 
-        var _animationsCount = 0;
-        var postAnimationCover = _.throttle(() => this.cover(), 100);
-        this.$target.on('transitionstart.snippet_editor, animationstart.snippet_editor', () => {
-            // We cannot rely on the fact each transition/animation start will
-            // trigger a transition/animation end as the element may be removed
-            // from the DOM before or it could simply be an infinite animation.
-            //
-            // By simplicity, for each start, we add a delayed operation that
-            // will decrease the animation counter after a fixed duration and
-            // do the post animation cover if none is registered anymore.
-            _animationsCount++;
-            setTimeout(() => {
-                if (!--_animationsCount) {
-                    postAnimationCover();
-                }
-            }, 500); // This delay have to be huge enough to take care of long
-                     // animations which will not trigger an animation end event
-                     // but if it is too small for some, this is the job of the
-                     // animation creator to manually ask for a re-cover
-        });
-        // On top of what is explained above, do the post animation cover for
-        // each detected transition/animation end so that the user does not see
-        // a flickering when not needed.
-        this.$target.on('transitionend.snippet_editor, animationend.snippet_editor', postAnimationCover);
+        this._animationsCount = 0;
+        this._animationTimeoutHandler = null;
+        this._postAnimationHandler = null;
+        this.$target.on(
+            'transitionstart.snippet_editor, animationstart.snippet_editor',
+            () => this._manageAnimationAndOverlay(1)
+        );
+        this.$target.on(
+            'transitionend.snippet_editor, animationend.snippet_editor',
+            () => this._manageAnimationAndOverlay(-1)
+        );
+        // Added cancel events as separate events as not yet well supported on every browser
+        // but worth adding for those where it is supported. It will avoid having the overlay
+        // reappearing when changing rapidly the animation.
+        this.$target.on(
+            'transitioncancel.snippet_editor',
+            () => this._manageAnimationAndOverlay(-1)
+        );
+        this.$target.on(
+            'animationcancel.snippet_editor',
+            () => this._manageAnimationAndOverlay(-1)
+        );
```

Comment:
In fact I am not sure we are going in the right way with this. If we have a animation starting, then a transitionstarting, then the transition end, then the animationend, it will not work on chrome/edge as the transition start event does not exist there. So it will do +1 -1 -1 -> = the overlay reappears too soon.

Reply:
I am using a separated counter from now and I have splitted all events in order to ensure that they get not activated because another selector is not supported. I know it does not make a beautiful code but at least it will work.

Reply:
I think we will not get something that fully works on all browsers, old and recent. But it is already better now like that and it will get better as users will be using more recent browsers.